lib/ansible/modules/network/avi/avi_pool.py

```diff
@@ -0,0 +1,338 @@
+
+
+def main():
+    argument_specs = dict(
+        state=dict(default='present',
+                   choices=['absent', 'present']),
+        a_pool=dict(type='str',),
+        ab_pool=dict(type='dict',),
+        ab_priority=dict(type='int',),
+        apic_epg_name=dict(type='str',),
+        application_persistence_profile_ref=dict(type='str',),
+        autoscale_launch_config_ref=dict(type='str',),
+        autoscale_networks=dict(type='list',),
+        autoscale_policy_ref=dict(type='str',),
+        capacity_estimation=dict(type='bool',),
+        capacity_estimation_ttfb_thresh=dict(type='int',),
+        cloud_config_cksum=dict(type='str',),
+        cloud_ref=dict(type='str',),
+        connection_ramp_duration=dict(type='int',),
+        created_by=dict(type='str',),
+        default_server_port=dict(type='int',),
+        description=dict(type='str',),
+        domain_name=dict(type='list',),
+        east_west=dict(type='bool',),
+        enabled=dict(type='bool',),
+        fail_action=dict(type='dict',),
+        fewest_tasks_feedback_delay=dict(type='int',),
+        graceful_disable_timeout=dict(type='int',),
+        health_monitor_refs=dict(type='list',),
+        host_check_enabled=dict(type='bool',),
+        inline_health_monitor=dict(type='bool',),
+        ipaddrgroup_ref=dict(type='str',),
+        lb_algorithm=dict(type='str',),
+        lb_algorithm_consistent_hash_hdr=dict(type='str',),
+        lb_algorithm_hash=dict(type='str',),
+        max_concurrent_connections_per_server=dict(type='int',),
+        max_conn_rate_per_server=dict(type='dict',),
+        name=dict(type='str',),
+        networks=dict(type='list',),
+        pki_profile_ref=dict(type='str',),
+        placement_networks=dict(type='list',),
+        prst_hdr_name=dict(type='str',),
+        request_queue_depth=dict(type='int',),
+        request_queue_enabled=dict(type='bool',),
+        rewrite_host_header_to_server_name=dict(type='bool',),
+        rewrite_host_header_to_sni=dict(type='bool',),
+        server_auto_scale=dict(type='bool',),
+        server_count=dict(type='int',),
+        server_name=dict(type='str',),
+        server_reselect=dict(type='dict',),
+        servers=dict(type='list',),
+        sni_enabled=dict(type='bool',),
+        ssl_key_and_certificate_ref=dict(type='str',),
+        ssl_profile_ref=dict(type='str',),
+        tenant_ref=dict(type='str',),
+        url=dict(type='str',),
+        use_service_port=dict(type='bool',),
+        uuid=dict(type='str',),
+        vrf_ref=dict(type='str',),
+    )
+    argument_specs.update(avi_common_argument_spec())
+    module = AnsibleModule(
+        argument_spec=argument_specs)
+    if not HAS_AVI:
+        return module.fail_json(msg=(
+            'Avi API SDK is not installed. '
+            'Use "sudo pip install --upgrade avisdk" to install Avi SDK. '
+            'For more details visit https://github.com/avinetworks/sdk.'))
+    return avi_ansible_api(module, 'pool',
```

Comment:
no error handling? we normally try to give 'nice' error messages to user, detailing the action attempted and giving as much information possible to assist them in fixing the issue. i.e `'failed to get current status of Pool: %s' % to_native(e)` when getting an exception while trying to contact the api to get a current status.